### PR TITLE
feat: Add ruby sqs dynamodb example

### PR DESCRIPTION
**Pull request by Pigius**
# Big picture behind

Currently, there aren't many examples of using the serverless approach with Ruby for AWS. Ruby is a great and mature language with a great community and lots of gems.

# General message

This is an example of the usage of the `SQS` with `DynamoDB`, `API Gateway`, and `AWS Lambda` functions, and `Cloudwatch` for monitoring. The service is for the purpose of creating lottery coupons for people. Each and every coupon consists of the `id`, `first_name`, `last_name` and `coupon_value` as well as of `created_at` timestamp. API Gateway is triggering producing an SQS message, and then the consumer will create a record within the DynamoDB table. The incoming requests are validated with the usage of [`dry-schema`](https://dry-rb.org/gems/dry-schema/1.5/).

The diagram could be found here:
aws-serverless-diagram

For the purpose of SQS part, the `Lift` [construct](https://github.com/getlift/lift) was used, to allow using `AWS CDK` constructs functionalities. 

Extra points:
- The readme is quite extensive, and I hope, that it will be easy to understand everything.
- Used the Ruby.27 runtime, as for the Ruby2.5 will be End of Support at the end of August. 
- The DynamoDB table name is a combination of service name and stage.

**Last important point regarding the example:** 
In the Lift, queue construct it is [mentioned](https://github.com/getlift/lift/blob/master/docs/queue.md#permissions) that we don't need to set up permission for the lambda. It's true, but if you will add extra IAM statements, it will not work anymore, and you need to manually add this to the `Serverless.yml`

**Update**: I've raised the [issue](https://github.com/getlift/lift/issues/47) within the Lift repository. The issue is due to deprecated usage of `provider.iamRoleStatements` in Lift.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,12 +13,14 @@ A collection of ready-to-deploy [Serverless Framework](https://github.com/server
 <details>
 <summary>Click to expand</summary>
 
-- [Getting Started](#getting-started)
```
Comment (Pigius) [inline README.md:16]: Not sure why it is changing, when I want to fix it, it does not allow me to save it. I believe due to `npm run docs`

Comment (pgrzesik) [reply README.md:16]: That's interesting, I'll see what might be the cause

File: aws-ruby-sqs-with-dynamodb/serverless.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+service: serverless-ruby-sqs-dynamodb
+
+frameworkVersion: '2'
+
+provider:
+  name: aws
+  profile: personal
```
Comment (pgrzesik) [inline aws-ruby-sqs-with-dynamodb/serverless.yml:7]: I think we can remove these two properties from the example

**Review by pgrzesik — COMMENTED** [review]
Thanks @Pigius - awesome work :clap: I only have very minor suggestion, otherwise we're good to go :100:

File: aws-ruby-sqs-with-dynamodb/serverless.yml (hunk 2 above)
Comment (Pigius) [reply aws-ruby-sqs-with-dynamodb/serverless.yml]: Agree, we don't need a profile here (in the readme it's information on how to do it, and for the region, a developer can provide the desired region, and if not - the default is `us-east-1`.

Fixed.

**Comment by Pigius** [comment]
@pgrzesik extra modification to the PR, As the https://www.npmjs.com/package/serverless-lift, has a new release 1.1.1 which included the fix for the https://github.com/getlift/lift/issues/47. So I've updated:

- the` lift` package to the newest one
- get rid of IAM permissions for the queue, as we don't need it anymore to be explicitly defined in the `serverless.yml`
- I've tested again the whole flow -> working flawlessly :)

**Review by pgrzesik — APPROVED** [review]
Thanks for addressing the issues + for additional updates :clap: Well done sir :bow:

**Comment by o5** [comment]
@Pigius Could you please share the link for diagram tool which you are used? It looks good 😊

**Comment by Pigius** [comment]
> @Pigius Could you please share the link for diagram tool which you are used? It looks good 😊

@o5 Hi, This is https://excalidraw.com/. It's free and open-source - it's brilliant! However, I recommend to always have a file with the `.excalidraw` extension somewhere locally to be able to edit the diagram continuously, as in our IT reality everything changes quite often :)

**Comment by Pigius** [comment]
@o5 Please also see https://libraries.excalidraw.com/?theme=light&sort=default#slobodan-aws-serverless as there are libraries with a great icons :)

**Comment by o5** [comment]
@Pigius Thanks! I used https://asciiflow.com which resolves the frequent changes but result from excalidraw looks better :)

**Comment by Pigius** [comment]
@o5 good to know, thanks!

**Pull request merged**